### PR TITLE
FourKitchens Release: Sprint 21

### PR DESCRIPTION
**Pull request by callinmullaney**
# READY FOR REVIEW

## Summary
- ~~SHS-148: Implement image optimize module to improve image compression pipelines and decrease file sizes across all sites~~
- SHS-4648: Fix a z-index issue with image caption appear above event date badges

## Need Review By (Date)
3/22

## Urgency
low

## Steps to Test
**SHS-148**:
1. The easiest way to test is to sync the EALC site `lando blt drupal:sync --site=ealc`
2. Go to `https://ealc.suhumsci.loc/events/east-asian-humanities-workshop-wu-mis-helen-troy-yiqun-zhou` and set the date as a future event.
3. Go to Screenshot from 2023-03-06 13-44-23
`https://ealc.suhumsci.loc/` and verify the caption appears below the event date badge.

**SHS-4648**:
~~1. Verify the automated build steps pass, properly install the `imageapi_optimize` and submodules, and import configuration.~~
~~2. This is easier to test on a remote environment because you can use external tools like `webpagetest.org` instead of chromes local lighthouse tool. Once deployed we ran `drush image:flush --all` twice and cleared caches.~~
~~3. Results of our testing were the follow. Tori and team found no noticeable differences in image quality when browsing the rest of the site.~~
   - Production: https://www.webpagetest.org/result/230228_AiDcZ9_BFD/ (image file sizes were 822kb)
   - Dev: https://www.webpagetest.org/result/230228_AiDcMN_BFF/ (image file sizes were 423kb)
   - Production: https://www.webpagetest.org/result/230307_BiDc3R_8RR/ (372kb)
   - Dev: https://www.webpagetest.org/result/230307_BiDc3H_8TY/ (242kb)
   - Production: https://www.webpagetest.org/result/230307_AiDcCS_9MR/ (219kb)
   - Dev: https://www.webpagetest.org/result/230307_AiDcBK_9MW/ (91kb)
~~4. Lighthouse (Google Page Speed Insights) was less dramatic but still and improvement. One thing we were able to configure with the `imageapi_optimize` is global WebP conversion added to the optimize pipeline. This hooked into the existing WebP module and didn't require us to add a new WebP filter to every image style. Note in the screenshot below the "Serve Images in next-gen formats" performance opportunity is no longer listed:~~
screenshot_from_2023-02-27_13-20-14

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -189,6 +189,9 @@
         "drupal/google_analytics": "^4.0",
         "drupal/honeypot": "^2.0",
         "drupal/hook_event_dispatcher": "~3.2.0",
+        "drupal/imageapi_optimize": "^4.0",
+        "drupal/imageapi_optimize_binaries": "^1.0@beta",
+        "drupal/imageapi_optimize_resmushit": "^2.0@beta",
```
Comment (pookmish) [inline composer.json]: What do  these do that imagemagick doesn't? These seems like just additional bloat

**Comment by callinmullaney** [comment]
@pookmish You're correct that the two additional modules were adding some overlapping functionality that already exists with imagemagik. I reworked this release to exclude those modules but kept the core `imageapi_optimize` present. The reason here is the existing WebP module only affects responsive images. However, there are many instances (mostly views) where a responsive image is not in use and an image style is used directly. With the `imageapi_optimize` I have a WebP pipeline available to assign to all image styles. With this default pipleline in place and the small changes to the image toolkit config we're still getting improved Google Page Speed scores with smaller image file sizes and images served in next-gen formats. 
Back to you!

**Review by pookmish — CHANGES_REQUESTED** [review]
This doesn't do anything to improve image performance other than decrease the derived quality by 10%. Sure you have a "pipeline" set up for a webp image, but the jpg/png is still loaded to the browser. 
Decreasing image quality is not a satisfactory solution to improve performance. Please revise.

**Comment by callinmullaney** [comment]
@pookmish closing this PR. Given the image pipeline isn't producing the expected results here we'll take a new approach that adds a convert filter for image styles directly. This will result in more configuration changes but should produce the desired markup output. 
We have another PR open and internally testing that adds a lot more responsive image styles and new basic image styles that are more specific for individual components. This is helping performance in a manner where we are no longer serving images in larger resolutions and down-scaling them with CSS: https://github.com/SU-HSDO/suhumsci/pull/1256
That being said, the image compression changes here are having a measurable affect on page sizes which directly affect page performance. I'd like to push forward with those specific adjustment as they do not visually affect the image quality but result in smaller files.

**Pull request closed without merging**